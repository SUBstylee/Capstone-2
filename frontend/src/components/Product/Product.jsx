import { Card } from "react-bootstrap";
import { Link } from "react-router-dom";
import './Product.scss';

import Rating from "../Rating/Rating";

import "./Product.scss";

const Product = ({ product }) => {
  return (
    <div className="Product">
    <Card className='h-100 my-2 p-1 rounded Product'>
      <Link to={`/products/${product._id}`}>
        <Card.Img className="cardImg" src={product.image} variant='top' />
      </Link>
      <Card.Body>
        <Link to={`/products/${product._id}`}>
          <Card.Title as='div'>
            <strong>{product.name}</strong>
          </Card.Title>
        </Link>
        <Card.Text as='div'>
          <Rating
            value={product.rating}
            text={`${product.numReviews} ${
              product.numReviews === 1 ? "review" : "reviews"
            }`}
          />
        </Card.Text>
        <Card.Text as='h3'>${product.price}</Card.Text>
      </Card.Body>
    </Card>
    </div>
  );
};

export default Product;
