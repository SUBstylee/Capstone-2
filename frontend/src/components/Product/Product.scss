.Product{
  box-shadow: 0px 2px 10px rgba($color: #505050, $alpha: 0.2);

  .cardImg {
    width: 100%;
    height: 20vw;
    object-fit: cover;
  };


  @media (max-width: 991px) {
    .cardImg {
      height: 30vw;
    };
  };
  @media (max-width: 767px) {
    .cardImg {
      height: 40vw;
    };
  };
  @media (max-width: 575px) {
    .cardImg {
      height: 70vw;
    };
  };
};