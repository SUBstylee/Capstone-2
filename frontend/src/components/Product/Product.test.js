import { render } from '@testing-library/react';
import Product from './Product';
import { BrowserRouter } from 'react-router-dom';

describe('test', () => {
    let product;
    beforeEach(() => {
        product = {
            rating: 5,
            numReviews: 1,
            price: 35.99,
            countInStock: 9,
            _id: "123",
            name: "Test Product",
            user: "456",
            image:
                "placeholder",
            category: "test",
            brand: "test brand",
            description:
                "This is a test.",
            reviews: [
                {
                    _id: "123456",
                    name: "John Doe",
                    rating: 5,
                    comment: "Test comment.",
                    user: "789",
                    createdAt: "2022-03-02T09:17:00.324Z",
                    updatedAt: "2022-03-02T09:17:00.324Z",
                },
            ],
            createdAt: "2021-09-02T06:30:21.695Z",
            updatedAt: "2021-09-09T17:44:57.014Z",
            __v: 1,
        };
    });

    test("renders Product component without crashing", () => {
        const { getByText } = render(
            <BrowserRouter>
                <Product product={product} />
            </BrowserRouter>
        );

        expect(getByText("Test Product")).toBeInTheDocument();
        // expect(getByText("1 reviews")).toBeInTheDocument();
        expect(getByText("$35.99")).toBeInTheDocument();
    });

    test("matches snapshot", function () {
        const { asFragment } = render(
            <BrowserRouter>
                <Product product={product} />
            </BrowserRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });

});