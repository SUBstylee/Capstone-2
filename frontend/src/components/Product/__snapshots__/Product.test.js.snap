// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Product"
  >
    <div
      class="h-100 my-2 p-1 rounded Product card"
    >
      <a
        href="/products/123"
      >
        <img
          class="card-img-top cardImg"
          src="placeholder"
        />
      </a>
      <div
        class="card-body"
      >
        <a
          href="/products/123"
        >
          <div
            class="card-title"
          >
            <strong>
              Test Product
            </strong>
          </div>
        </a>
        <div
          class="card-text"
        >
          <div
            class="Rating"
          >
            <span>
              <i
                class="fa-solid fa-star"
                style="color: rgb(238, 188, 29);"
              />
            </span>
            <span>
              <i
                class="fa-solid fa-star"
                style="color: rgb(238, 188, 29);"
              />
            </span>
            <span>
              <i
                class="fa-solid fa-star"
                style="color: rgb(238, 188, 29);"
              />
            </span>
            <span>
              <i
                class="fa-solid fa-star"
                style="color: rgb(238, 188, 29);"
              />
            </span>
            <span>
              <i
                class="fa-solid fa-star"
                style="color: rgb(238, 188, 29);"
              />
            </span>
            <span
              class="ps-2"
            >
              1 review
            </span>
          </div>
        </div>
        <h3
          class="card-text"
        >
          $35.99
        </h3>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
